frontstage/controllers/case_controller.py

```diff
@@ -64,42 +32,27 @@ def calculate_case_status(case, collection_instrument_type):
     elif case_group_status == 'INPROGRESS' and collection_instrument_type == 'SEFT':
         status = 'Downloaded'
 
-    logger.debug('Retrieved the status of case', case=case['id'], status=status)
+    logger.debug('Retrieved the status of case', collection_instrument_type=collection_instrument_type,
+                 status=status)
     return status
 
 
-def case_is_enrolled(case, respondent_id):
-    logger.debug('Checking status of case for respondent', party_id=respondent_id)
-    association = next((association
-                       for association in case['business_party']['associations']
-                       if association['partyId'] == respondent_id), {})
-    enrolment_status = next((enrolment['enrolmentStatus']
-                            for enrolment in association.get('enrolments', [])
-                            if enrolment['surveyId'] == case['survey']['id']), '')
-    if not enrolment_status:
-        logger.warning('No status found for case', party_id=respondent_id)
-    return enrolment_status == 'ENABLED'
-
-
-def check_case_permissions(party_id, case_party_id, case_id=None):
-    logger.debug('Party requesting access to case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
-
-    if party_id != case_party_id:
-        raise NoSurveyPermission(party_id, case_id, case_party_id)
-
-    logger.debug('Party has permission to access case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
+def check_case_permissions(party_id, case_id, business_party_id, survey_short_name):
+    logger.debug('Party requesting access to case', party_id=party_id, case_id=case_id,
+                 business_party_id=business_party_id, survey_short_name=survey_short_name)
+    match = False
 
+    party = party_controller.get_respondent_party_by_id(party_id)
+    for association in party['associations']:
+        if association['partyId'] == business_party_id:
+            survey = survey_controller.get_survey_by_short_name(survey_short_name)
+            for enrolment in association['enrolments']:
+                if enrolment['surveyId'] == survey['id']:
+                    match = True
```

Comment:
It's a bit difficult to understand what's happening here. Could this be broken down in a similar way to the survey list code with list comprehensions/generators